README: Add SASL section (#512)
segmentio/kafka-go: README.md

	Dialer:   dialer,
})
```

## SASL Support

> For a bare bones Conn type or in the Reader/Writer configs you can specify a dialer option for SASL authentication. If the `SASLMechanism` field is `nil`, it will not authenticate with SASL.

Reviewer: What about something like this?
```suggestion
You can specify an option on the `Dialer` to use SASL authentication.  The `Dialer` can be used directly to open a `Conn` or it can be passed to a `Reader` or `Writer` via their respective configs.  If the `SASLMechanism` field is `nil`, it will not authenticate with SASL.
```
I think it aligns a little bit more with the code since, in all cases, you configure the mechanism on the dialer.

Definitely open to further edits. 😄

There are implementations for [Plain](https://godoc.org/github.com/segmentio/kafka-go/sasl/plain#Mechanism)
and [SCRAM](https://godoc.org/github.com/segmentio/kafka-go/sasl/scram#Mechanism) included.

> ### Connection

Reviewer: Do we want any examples of `Basic`?

Author: do you mean examples of plain (since i only included scram?) or something else

Reviewer: Ah yes, sorry.  `Plain` is what I meant. :)

Author: added a section "SASL Authentication Types"

```go

> mechanism, err := scram.Mechanism(...config...)

Reviewer 2: ```suggestion
mechanism, err := scram.Mechanism(...)
```

if err != nil {
    panic(err)
}

dialer := &kafka.Dialer{
    Timeout:       10 * time.Second,
    DualStack:     true,
    SASLMechanism: mechanism,
}

conn, err := dialer.DialContext(ctx, "tcp", "localhost:9093")
```

### Reader

```go

> mechanism, err := scram.Mechanism(...config...)

Reviewer 2: ```suggestion
mechanism, err := scram.Mechanism(...)
```

if err != nil {
    panic(err)
}

dialer := &kafka.Dialer{
    Timeout:       10 * time.Second,
    DualStack:     true,
    SASLMechanism: mechanism,
}

r := kafka.NewReader(kafka.ReaderConfig{
    Brokers:        []string{"localhost:9093"},
    GroupID:        "consumer-group-id",
    Topic:          "topic-A",
    Dialer:         dialer,
})
```

### Writer

```go

> mechanism, err := scram.Mechanism(...config...)

Reviewer 2: The `config...` got me confused, it looked like a slice expansion. How about the simpler:
```suggestion
mechanism, err := scram.Mechanism(...)
```

Reviewer: Another option would be to populate with sample arguments:
```suggestion
mechanism, err := scram.Mechanism(scram.SHA512, "username", "password")
```

Author: yeah it did look suspiciously language-like, i think i'll do sample args
